### ISSUE TITLE
ivre db2view nmap throws an exception "pymongo.errors.OperationFailure"

### ISSUE THREAD

**Issue by nibiwodong** [cea-sec/ivre#909]
### IVRE Version
```
$ ivre version
root@a960f41c08fd:/# ivre --version
IVRE - Network recon framework
Copyright 2011 - 2020 Pierre LALET <pierre@droids-corp.org>
Version 9bb175ff

Python 2.7.16 (default, Oct 10 2019, 22:02:15) 
[GCC 8.3.0]

Linux a960f41c08fd 3.10.0-514.el7.x86_64 #1 SMP Tue Nov 22 16:42:41 UTC 2016 x86_64

Dependencies:
    Python module Crypto: 2.6.1
    Python module pymongo: 3.7.1
    Python module py2neo: missing
    Python module sqlalchemy: missing
    Python module psycopg2: missing
    Python module krbV: missing
    Python module PIL: 5.4.1
    Python module MySQLdb: missing
    Python module dbus: missing
    Python module matplotlib: missing
    Python module bottle: 0.12.15
    Python module OpenSSL: missing
    Python module tinydb: missing

```

### Summary
ivre db2view nmap throws an exception "pymongo.errors.OperationFailure"

### Expected behavior
docker exec -it ivreclient ivre db2view nmap
this command should transfer hosts collection data to views collection.

### Actual behavior
When I have scanned 2000 hosts,  I can not execute ivre db2view nmap. 
This is the logs:
```
[root@localhost ~]# docker exec -it ivreclient ivre db2view nmap
Traceback (most recent call last):
  File "/usr/local/bin/ivre", line 88, in <module>
    main()
  File "/usr/local/bin/ivre", line 60, in main
    tools.get_command(next(iter(possible_commands)))()
  File "/usr/local/lib/python2.7/dist-packages/ivre/tools/db2view.py", line 130, in main
    for elt in itr:
  File "/usr/local/lib/python2.7/dist-packages/ivre/view.py", line 440, in to_view
    next_recs.append(next(itrs[i]))
  File "/usr/local/lib/python2.7/dist-packages/ivre/view.py", line 390, in from_nmap
    for rec in db.nmap.get(flt, sort=[("addr", 1)]):
  File "/usr/local/lib/python2.7/dist-packages/ivre/db/mongo.py", line 1476, in get
    for host in self._get(spec, **kargs):
  File "/usr/lib/python2.7/dist-packages/pymongo/cursor.py", line 1189, in next
    if len(self.__data) or self._refresh():
  File "/usr/lib/python2.7/dist-packages/pymongo/cursor.py", line 1104, in _refresh
    self.__send_message(q)
  File "/usr/lib/python2.7/dist-packages/pymongo/cursor.py", line 982, in __send_message
    helpers._check_command_response(first)
  File "/usr/lib/python2.7/dist-packages/pymongo/helpers.py", line 155, in _check_command_response
    raise OperationFailure(msg % errmsg, code, response)
pymongo.errors.OperationFailure: Executor error during find command :: caused by :: Sort operation used more than the maximum 33554432 bytes of RAM. Add an index, or specify a smaller limit.
```

So I increased the mongodb RAM from32MB to 990MB in mongdb container. It works, but when hosts collection become bigger, this exception come again.
```
use admin
db.adminCommand({setParameter: 1, internalQueryExecMaxBlockingSortBytes:1048575990})
```

```
[root@localhost ~]# docker exec -it ivreclient ivre db2view nmap
Traceback (most recent call last):
  File "/usr/local/bin/ivre", line 88, in <module>
    main()
  File "/usr/local/bin/ivre", line 60, in main
    tools.get_command(next(iter(possible_commands)))()
  File "/usr/local/lib/python2.7/dist-packages/ivre/tools/db2view.py", line 130, in main
    for elt in itr:
  File "/usr/local/lib/python2.7/dist-packages/ivre/view.py", line 440, in to_view
    next_recs.append(next(itrs[i]))
  File "/usr/local/lib/python2.7/dist-packages/ivre/view.py", line 390, in from_nmap
    for rec in db.nmap.get(flt, sort=[("addr", 1)]):
  File "/usr/local/lib/python2.7/dist-packages/ivre/db/mongo.py", line 1476, in get
    for host in self._get(spec, **kargs):
  File "/usr/lib/python2.7/dist-packages/pymongo/cursor.py", line 1189, in next
    if len(self.__data) or self._refresh():
  File "/usr/lib/python2.7/dist-packages/pymongo/cursor.py", line 1104, in _refresh
    self.__send_message(q)
  File "/usr/lib/python2.7/dist-packages/pymongo/cursor.py", line 982, in __send_message
    helpers._check_command_response(first)
  File "/usr/lib/python2.7/dist-packages/pymongo/helpers.py", line 155, in _check_command_response
    raise OperationFailure(msg % errmsg, code, response)
pymongo.errors.OperationFailure: Executor error during find command :: caused by :: Sort operation used more than the maximum 1048575990 bytes of RAM. Add an index, or specify a smaller limit.
```

### How to reproduce
make sure hosts collection is big enough, maybe 2000 hosts or more.

**Comment by p-l-** [comment]
Hi,

Thanks for your report. Even with much bigger collections than 2k hosts, I cannot reproduce this issue.
- Could you confirm that you do not use any filter (like `ivre db2view nmap --country US` for example)?
- Could you also confirm that you properly created the indexes using `ivre scancli --init` prior to inserting scan data with `ivre scan2db`? 
We will figure out a fix!

**Comment by nibiwodong** [comment]
I do not using `ivre scancli --init`. Is this a necessary step?

There is a lot of data in the database. If I execute `ivre scancli --init` , the data will be cleared.

What should I do now?

**Comment by nibiwodong** [comment]
This problem has been solved.
I modified mongo.py
![image](https://user-images.githubusercontent.com/17465789/77717376-4149b980-701b-11ea-9418-e1667ac26fd4.png)

and then `ivre scancli --init`
the hosts indexSizes is 1(only _id_), now is 32.
![image](https://user-images.githubusercontent.com/17465789/77717416-63dbd280-701b-11ea-8977-8b0366ce0cb6.png)

**Comment by p-l-** [comment]
That's really weird. Did you get any errors when running `ivre scancli --init`?

**Comment by nibiwodong** [comment]
`ivre scancli --ensure-indexes ` is a better choise, this option will not clear mongodb hosts collection.
--ensure-indexes      Create missing indexes (will lock the database).
Am I right?

**Comment by p-l-** [comment]
Yes, you are right. But I would like to know if, when you first created the database with `ivre scancli --init`, you got errors? Oh, first, did you, **before inserting any data**, run `ivre scancli --init`?

**Comment by nibiwodong** [comment]
I did not run `ivre scancli --init` before insert any data, So these is no index in hosts collection.
And after I modified mongo.py like this:
![image](https://user-images.githubusercontent.com/17465789/78095048-99f8c800-7408-11ea-80cc-216407d3b9a3.png)
This will only run create_indexes(). To my surprise, when I run `ivre scancli --init`, the data in hosts collection was cleared. Is this normal?
So I have to scan again  :(

I think there are some differences between create_indexes() and ensure_indexes(). But I am not sure what difference is .
